### PR TITLE
Update backend to use channel, and not depend on glib

### PR DESCRIPTION
**Pull request by ids1024**
The glib based version isn't easily usable in a multi-threaded context, or outside of a glib mainloop. This works for the GTK UI, but is also suitable for building an Iced UI on top of. It may be helpful for implementing a command line interface too.

This changes a lot of code, but behavior should generally be unchanged.

### CODE REVIEW

**Comment by ids1024** [comment]
This has some conflicts. I'll rebase it after https://github.com/pop-os/keyboard-configurator/pull/175 is merged.

It would be good to get this merged before anything else adds conflicts, though it's otherwise not urgent since this shouldn't change behavior. Having it unmerged is sufficient for any prototyping of an Iced UI.

**Review by n3m0-22 — APPROVED** [review]
Tested on `lemp11`(pop), `gaze15`(pop), `oryp4`(win10 and pop)

Tested with Launch Heavy and built in keyboard on lemp11 and gaze15:
Keyboard Configurator, appimage, msi

Everything is working as expected. I have seen no issues or regressions. Looks good.

File: backend/examples/daemon.rs
Comment (13r0ck) [inline backend/examples/daemon.rs:18]: The bool is no longer necessary when creating the backend. ( #175 )

**Review by 13r0ck — COMMENTED** [review]
Seem to be quite a lot of `unwrap`s, would it be possible to handle those errors a bit better?

**Comment by ids1024** [comment]
It's pretty much standard to use `.unwrap()` with things like `Mutex::lock`. Are there any specific `.unwrap()` calls that seem problematic?

File: backend/examples/daemon.rs
Comment (ids1024) [reply backend/examples/daemon.rs]: Good catch. Basically all other code here is tested in CI, but this one wasn't since examples/tests are only run on Linux.

**Review by n3m0-22 — CHANGES_REQUESTED** [review]
With permissions set to run as a program the new appimage will not run by clicking on it. I am no longer seeing the prompt for Authentication Required. It will run from the terminal with and without sudo. When running from the terminal I am seeing `Gtk-Message: 07:56:36.220: Failed to load module "canberra-gtk-module"` but this is the same with the previous appimage tested yesterday.

File: backend/src/board.rs
Comment (13r0ck) [inline backend/src/board.rs:106]: Ya you are right about unwrapping a mutex. In situations like this where the function return is already a `Result_, String>` returning a result is easy enough that I don't think we should have unwrap calls here,

**Comment by ids1024** [comment]
@13r0ck Looking through the diff, it looks like the only unwraps added are for mutex locks, `OnceCell`s, or `ThreadPool` creation. It's typical to unwrap in both cases. The `OnceCell` unwraps should be statically impossible to fail, and the mutex locks will only fail if another thread panics while holding the mutex. Then there's one unwrap for `ThreadPool` creation, but this only fails if the OS fails to create a thread (the standard library `std::thread::spawn` function just panics on this, so unwrapping here is normal).

File: backend/src/board.rs
Comment (13r0ck) [reply backend/src/board.rs:106]: But on second review most situations like this are not a result of this PR.

Comment (13r0ck) [reply backend/src/board.rs:106]: I will approve if you want to, but seems like a simple enough change

**Comment by n3m0-22** [comment]
The appimage is now working as expected. Testing the rest now.

**Review by n3m0-22 — APPROVED** [review]
Tested again with the Launch Heavy on Pop, Win10, and MacOS as well as built in keyboards on the gaze15 and lemp11. Everything seems to be working as expected now. Looks good.

File: backend/src/board.rs
Comment (ids1024) [reply backend/src/board.rs:106]: I'd say panicking is the correct behavior if the `Layout` is invalid. `layout` here can only ever be one of the layouts compiled into the library. Libraries should not attempt to return errors for things that are internal bugs in the library itself.

Comment (13r0ck) [reply backend/src/board.rs:106]: I disagree. 
If a function returns a result if the function is called incorrectly, then it should return an `Err`, not panic, that is the point of the `Result` enum.

**Review by 13r0ck — APPROVED** [review]
I strongly disagree with the assessment that a function with a `Result` return type should ever panic. 

But I have not interest in bickering any more.

**Pull request merged**
